app/components/PlatformCards/GtcStakingPlatform.tsx

```diff
@@ -0,0 +1,143 @@
+// --- Methods
+import React, { useContext, useEffect, useState } from "react";
+
+// --- Datadog
+import { datadogLogs } from "@datadog/browser-logs";
+
+// --- Identity tools
+import {
+  Stamp,
+  PLATFORM_ID,
+  PROVIDER_ID,
+  VerifiableCredential,
+  CredentialResponseBody,
+  VerifiableCredentialRecord,
+} from "@gitcoin/passport-types";
+import { fetchVerifiableCredential } from "@gitcoin/passport-identity/dist/commonjs/src/credentials";
+
+// --- Style Components
+import { SideBarContent } from "../SideBarContent";
+import { DoneToastContent } from "../DoneToastContent";
+import { useToast } from "@chakra-ui/react";
+
+// --- Context
+import { CeramicContext } from "../../context/ceramicContext";
+import { UserContext } from "../../context/userContext";
+
+// --- Platform definitions
+import { getPlatformSpec } from "../../config/platforms";
+import { STAMP_PROVIDERS } from "../../config/providers";
+
+const iamUrl = process.env.NEXT_PUBLIC_PASSPORT_IAM_URL || "";
+
+// Each provider is recognised by its ID
+const platformId: PLATFORM_ID = "GtcStaking";
+
+export default function GtcStakingPlatform(): JSX.Element {
+  const { address, signer } = useContext(UserContext);
+  const { handleAddStamps, allProvidersState } = useContext(CeramicContext);
+  const [isLoading, setLoading] = useState(false);
+  const [canSubmit, setCanSubmit] = useState(false);
+
+  // find all providerIds
+  const providerIds =
+    STAMP_PROVIDERS[platformId]?.reduce((all, stamp) => {
+      return all.concat(stamp.providers?.map((provider) => provider.name as PROVIDER_ID));
+    }, [] as PROVIDER_ID[]) || [];
```

Comment:
This can be memoized (check the changes on this commit: https://github.com/gitcoinco/passport/commit/933b27296ba6123a42938815549f762e53817b5a)